Model/Payment/Gateway/Command/RefundPayment.php

```diff
@@ -59,6 +67,11 @@ public function execute(array $commandSubject): void
         }
         $orderExtensionData->setIsRefundInProgress(true);
         $this->orderExtensionDataRepository->save($orderExtensionData);
+        // The save() call above doesn't commit the change because Magento starts a transaction in CreditmemoService
+        // before calling this function. We have to manually commit the change to the database so that the value is
+        // set before calling Bold Checkout which calls back to the Magento module to update payments.
+        $this->connection->getConnection()->commit()->beginTransaction();
```

Comment:
I'm concerned that manually triggering a `commit()` here could cause other unforeseen issues, especially with custom/third-party extensions.